tests/e2e/testsLibrary/CodeExecutionTests.ts

```diff
@@ -8,22 +8,45 @@
  * SPDX-License-Identifier: EPL-2.0
  **********************************************************************/
 
-import { CLASSES, Terminal, TopMenu, Ide } from '..';
+import { CLASSES, Terminal, TopMenu, Ide, DialogWindow } from '..';
 import { e2eContainer } from '../inversify.config';
 
 const terminal: Terminal = e2eContainer.get(CLASSES.Terminal);
 const topMenu: TopMenu = e2eContainer.get(CLASSES.TopMenu);
 const ide: Ide = e2eContainer.get(CLASSES.Ide);
+const dialogWindow: DialogWindow = e2eContainer.get(CLASSES.DialogWindow);
+
+const dialogWindowOpenLinkButtonText: string = 'Open Link';
+const dialogWindowCloseButtonText: string = 'close';
 
 export function runTask(taskName: string, timeout: number) {
-    test( `Run command '${taskName}'`, async () => {
+    test(`Run command '${taskName}'`, async () => {
         await topMenu.runTask(taskName);
         await ide.waitNotification('has exited with code 0.', timeout);
     });
 }
 
+export function runTaskWithDialogShellAndOpenLink(taskName: string, expectedDialogText: string, timeout: number) {
+    test(`Run command '${taskName}' expecting dialog shell`, async () => {
+        await topMenu.runTask(taskName);
+        await dialogWindow.waitDialog(timeout, expectedDialogText);
```

Comment:
replace 
```
        await dialogWindow.waitDialog(timeout, expectedDialogText);
        await dialogWindow.clickToButton(dialogWindowOpenLinkButtonText);
        await dialogWindow.waitDialogDissappearance();
```
by
```
await dialogWindow.waitDialogAndOpenLink(timeout, expectedDialogText);
```

Reply:
https://github.com/eclipse/che/blob/master/tests/e2e/pageobjects/ide/DialogWindow.ts#L79 this and the method above it seem to be in an incorrect order

Reply:
I've fixed the method:
From:
```
        await this.ide.waitApllicationIsReady(await this.getApplicationUrlFromDialog(dialogText), timeout);
        await this.clickToOpenLinkButton();
```
To:
```
        const applicationUrlFromDialog: string = await this.getApplicationUrlFromDialog(dialogText);
        await this.clickToOpenLinkButton();
        await this.ide.waitApllicationIsReady(applicationUrlFromDialog, timeout);
```
This resolved the race condition